library/agent/safeCreateRegExp.test.ts

```diff
@@ -0,0 +1,24 @@
+import * as t from "tap";
+import { safeCreateRegExp } from "./safeCreateRegExp";
+
+t.test("safeCreateRegExp returns RegExp for valid pattern", async (t) => {
+  const regex = safeCreateRegExp("abc", "i");
+  t.ok(regex instanceof RegExp);
+  t.match("Abc", regex);
+});
+
+t.test("safeCreateRegExp returns undefined for invalid pattern", async (t) => {
+  const regex = safeCreateRegExp("[", "i");
+  t.equal(regex, undefined);
+});
+
+t.test("safeCreateRegExp returns undefined for invalid flags", async (t) => {
+  const regex = safeCreateRegExp("abc", "invalidflag");
+  t.equal(regex, undefined);
+});
+
+t.test("safeCreateRegExp works with empty pattern", async (t) => {
+  const regex = safeCreateRegExp("", "");
+  t.ok(regex instanceof RegExp);
+  t.match("", regex);
```

Comment:
do we have checks for blockedUserAgents/monitoredUserAgents that the result is not allowed to be `""`? We have those in Java but wasn't sure here?

Reply:
`!""` okay nvm that's already there :))

Reply:
https://github.com/AikidoSec/firewall-node/pull/601/commits/eabf728f9d7f49754a4d632da4f529c70a4fb8cc